src/tmx_json.c

```diff
@@ -205,39 +207,164 @@ static int pjson_tile(json_t *tile_el, tmx_tile **tile_headaddr) {
 	return 1;
 }
 
+
+/* This is a rather slow solution - anyone with more spare time, feel free to rework the whole structure of
+   the tiles list to make this faster */

```

Comment:
I've been looking into this just now. Is there any implementation of a hash table in the standard C library? I can't use std::set (or anything like that) since that is C++ code. Adding an external library for hash tables would be a bit over the top. And implementing my own hash table class in C seems difficult, too, unless I'm overcomplicating things and it's actually really easy. I could give it a try, but I don't want to bloat the code too much, so maybe there is a simple and light-weight way to implement a hash table in C?